src/main/java/edu/cornell/kfs/coa/batch/businessobject/WorkdayOpenAccountDetail.java

```diff
@@ -0,0 +1,212 @@
+        this.subFundGroupWageIndicator = subFundGroupWageIndicator;
+    }
+
+    public String getSubFundGroupCode() {
+        return subFundGroupCode;
+    }
+
+    public void setSubFundGroupCode(String subFundGroupCode) {
+        this.subFundGroupCode = subFundGroupCode;
+    }
+
+    public String getHigherEdFunctionCode() {
+        return higherEdFunctionCode;
+    }
+
+    public void setHigherEdFunctionCode(String higherEdFunctionCode) {
+        this.higherEdFunctionCode = higherEdFunctionCode;
+    }
+
+    public Date getAccountEffectiveDate() {
+        return accountEffectiveDate;
+    }
+
+    public void setAccountEffectiveDate(Date accountEffectiveDate) {
+        this.accountEffectiveDate = accountEffectiveDate;
+    }
+
+    public String getAccountEffectiveDateString() {
+        return accountEffectiveDateString;
+    }
+
+    public void setAccountEffectiveDateString(String accountEffectiveDateString) {
+        this.accountEffectiveDateString = accountEffectiveDateString;
+    }
+
+    public String getAccountClosedIndicator() {
+        return accountClosedIndicator;
+    }
+
+    public void setAccountClosedIndicator(String accountClosedIndicator) {
+        this.accountClosedIndicator = accountClosedIndicator;
+    }
+
+    public String getAccountTypeCode() {
+        return accountTypeCode;
+    }
+
+    public void setAccountTypeCode(String accountTypeCode) {
+        this.accountTypeCode = accountTypeCode;
+    }
+
+    public String getSubAccountNumber() {
+        return subAccountNumber;
+    }
+
+    public void setSubAccountNumber(String subAccountNumber) {
+        this.subAccountNumber = subAccountNumber;
+    }
+
+    public String getSubAccountName() {
+        return subAccountName;
+    }
+
+    public void setSubAccountName(String subAccountName) {
+        this.subAccountName = subAccountName;
+    }
+
+    public String getSubAccountActiveIndicator() {
+        return subAccountActiveIndicator;
+    }
+
+    public void setSubAccountActiveIndicator(String subAccountActiveIndicator) {
+        this.subAccountActiveIndicator = subAccountActiveIndicator;
+    }
+
+    public String getObjectCode() {
+        return objectCode;
+    }
+
+    public void setObjectCode(String objectCode) {
+        this.objectCode = objectCode;
+    }
+
+    public String getSubObjectCode() {
+        return subObjectCode;
+    }
+
+    public void setSubObjectCode(String subObjectCode) {
+        this.subObjectCode = subObjectCode;
+    }
+
+    public String getSubObjectName() {
+        return subObjectName;
+    }
+
+    public void setSubObjectName(String subObjectName) {
+        this.subObjectName = subObjectName;
+    }
+    
+    public boolean isHeaderDetailRow() {
+        return headerDetailRow;
+    }
+
+    public void setHeaderDetailRow(boolean headerDetailRow) {
+        this.headerDetailRow = headerDetailRow;
+    }
+
+    public String toCsvString() {
```

Comment:
I was reviewing CSVWriter examples, I don't see where it supports the custom logic I have to build the header row, in particular the quotes versus no quotes and the effective date formatted date versus just the string effective date

Reply:
Your custom logic might be able to be accommodated. (Using the CSVWriter class appeals to me, so I'd like to see it work.)  What about something with logic like this?


```
public String toCsvString() {
 String csv;
  if (headerDetailRow) {
    String[] rowData = {chart, accountNumber, ...,  accountEffectiveDateString, ...}
    try (StringWriter sw = new StringWriter(); CSVWriter csvWriter = new CSVWriter(sw, CSVWriter.DEFAULT_SEPARATOR, CSVWriter.NO_QUOTE_CHARACTER, CSVWriter.DEFAULT_ESCAPE_CHARACTER, CSVWriter.DEFAULT_LINE_END)) {
      csvWriter.writeNext(rowData);
      csv = sw.toString();
    }
  } else {
    SimpleDateFormat sdf = new SimpleDateFormat(KFSConstants.MONTH_DAY_YEAR_DATE_FORMAT, Locale.US);
    String[] rowData = {chart, accountNumber, ...,  sdf.format(accountEffectiveDate), ...}
    try (StringWriter sw = new StringWriter(); CSVWriter csvWriter = new CSVWriter(sw)) {
      csvWriter.writeNext(rowData);
      csv = sw.toString();
    }
  }
  return csv
}
```

Reply:
Based on your code, it looks like only the header row is omitting the quotes. Is there a specific requirement to leave the header labels unquoted? I wouldn't think quoted headers would be an issue, unless the Workday folks specifically need it that way.

As for the formatted-date-vs-string-date part, that also looks like a headerRow-vs-dataRow situation based on your code. After looking a little further, you may be able to get the functionality you need from the OpenCSV library (which has the CSVWriter), but rather than using the CSVWriter directly, perhaps you could use the StatefulBeanToCsv class instead. The latter provides a simpler way of mapping a bean/POJO to a CSV row.

There's an example of how to use this in the PaymentWorksUploadSuppliersServiceImpl class. In addition, we have a custom EnumConfiguredMappingStrategy class, which allows for using a Java enum to configure which object properties to write and what labels to use for them in the header row. A setup similar to this should allow you to get the output you need, though you may need a custom getter on the DTO that will return the date as a formatted string.